chainermn/functions/__init__.py

```diff
@@ -0,0 +1,2 @@
+from chainermn.functions.communication import Recv  # NOQA
+from chainermn.functions.communication import Send  # NOQA
```

Comment:
Module name `communication` seems too general to me, as ChainerMN is almost all about communication 😉. How about thinking more specific names, such as `point_to_point_communication` ?